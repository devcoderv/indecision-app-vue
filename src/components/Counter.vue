<template>
  <h1>Counter</h1>
  <p>{{ counter }}<sup>2</sup> = {{ squerCounter }}</p>
  <div>
    <button @click="decrement">-1</button>
    <button @click="increment">+1</button>
  </div>
</template>

<script>
export default {
  props: {
    title: String,
    start: {
      type: Number,
      default: 1,
      validator(value) {
        return value >= 0;
      },
    },
  },
  data() {
    return {
      counter: this.start,
    };
  },
  computed: {
    squerCounter() {
      return this.counter * this.counter;
    },
  },
  methods: {
    increment() {
      this.counter++;
    },
    decrement() {
      this.counter--;
    },
  },
};
</script>
