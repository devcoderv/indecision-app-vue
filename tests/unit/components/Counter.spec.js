import {shallowMount} from '@vue/test-utils'
import Counter from '@/components/Counter'

describe('Counter Component', () => {
  

  test('debe de hacer match con el snapshot', () => {
    
    // const wrapper = shallowMount(Counter)
    
    // expect(wrapper.html()).toMatchSnapshot()
    
    const wrapper = shallowMount(Counter)

    const h2Value = wrapper.find('h1').text()

    expect(h2Value).toBe('Counter')
  
  
  })

  test('El valor por defecto debe de ser 100 en el ultimo p', () => {
    
    // const wrapper = shallowMount(Counter)
    
    // expect(wrapper.html()).toMatchSnapshot()
    
    const wrapper = shallowMount(Counter)

    const value = wrapper.find('[data-testid="counter"]').text()

    expect(value).toBe('100')
  
  
  })
  


})
