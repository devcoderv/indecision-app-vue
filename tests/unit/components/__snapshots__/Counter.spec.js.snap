// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter Component debe de hacer match con el snapshot 1`] = `
<h1>Counter</h1>
<p>1<sup>2</sup> = 1</p>
<div><button>-1</button><button>+1</button></div>
`;
